glasgow: 0-unstable-2025-01-26 -> 0-unstable-2025-07-28

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...19fc504f5b745e3b487c342cb96e3a2455e890e4)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-02-12 in filename of file in /nix/store/bh7qva9rrxw1rfzq56kfvjmpwbdhp3d2-glasgow-0-unstable-2025-02-12

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/bh7qva9rrxw1rfzq56kfvjmpwbdhp3d2-glasgow-0-unstable-2025-02-12 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/7c73e144a8f9952dcafaa9a3159ae5e52be42e19.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/7c73e144a8f9952dcafaa9a3159ae5e52be42e19#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/bh7qva9rrxw1rfzq56kfvjmpwbdhp3d2-glasgow-0-unstable-2025-02-12
ls -la /nix/store/bh7qva9rrxw1rfzq56kfvjmpwbdhp3d2-glasgow-0-unstable-2025-02-12/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NickCao reviewed (commented):
Does `realVersion` require an update?

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...2ebe547c07ea9f796d0134733e5a3095fe91d113)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-02-21 in filename of file in /nix/store/n2sm1ds07vmap7md062iwd8z19wkdz7y-glasgow-0-unstable-2025-02-21

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/n2sm1ds07vmap7md062iwd8z19wkdz7y-glasgow-0-unstable-2025-02-21 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/ed06f267a160096733ec8c65fe5d68c94a5cdd77.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/ed06f267a160096733ec8c65fe5d68c94a5cdd77#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/n2sm1ds07vmap7md062iwd8z19wkdz7y-glasgow-0-unstable-2025-02-21
ls -la /nix/store/n2sm1ds07vmap7md062iwd8z19wkdz7y-glasgow-0-unstable-2025-02-21/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...1cd7d021d40b8504547dfe4cc503fd6314a7c2e2)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-03-13 in filename of file in /nix/store/1f0a563cyl5wbc0qy9ykk30534wifi7a-glasgow-0-unstable-2025-03-13

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/1f0a563cyl5wbc0qy9ykk30534wifi7a-glasgow-0-unstable-2025-03-13 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/8a7944c5c49eff9c6f26f2ca88c382e8c394341c.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/8a7944c5c49eff9c6f26f2ca88c382e8c394341c#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/1f0a563cyl5wbc0qy9ykk30534wifi7a-glasgow-0-unstable-2025-03-13
ls -la /nix/store/1f0a563cyl5wbc0qy9ykk30534wifi7a-glasgow-0-unstable-2025-03-13/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...a41759f3f94f9a802e114b322e2c890f8309fcc7)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-03-23 in filename of file in /nix/store/s7pryl2rzmnyfk22cb9d5wvwicbcl02z-glasgow-0-unstable-2025-03-23

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/s7pryl2rzmnyfk22cb9d5wvwicbcl02z-glasgow-0-unstable-2025-03-23 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/773c5bd60f944bea0f446e2fe0ba17b8b857b4c5.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/773c5bd60f944bea0f446e2fe0ba17b8b857b4c5#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/s7pryl2rzmnyfk22cb9d5wvwicbcl02z-glasgow-0-unstable-2025-03-23
ls -la /nix/store/s7pryl2rzmnyfk22cb9d5wvwicbcl02z-glasgow-0-unstable-2025-03-23/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...3d7b185889544076afb09cf0c3b0c193a6f3699f)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-04-04 in filename of file in /nix/store/7g9lwybbndzxrhanblrrk2ndqgdagqqq-glasgow-0-unstable-2025-04-04

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/7g9lwybbndzxrhanblrrk2ndqgdagqqq-glasgow-0-unstable-2025-04-04 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/e24d86583c7ef1bbfab91ba8250384a27644f37b.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/e24d86583c7ef1bbfab91ba8250384a27644f37b#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/7g9lwybbndzxrhanblrrk2ndqgdagqqq-glasgow-0-unstable-2025-04-04
ls -la /nix/store/7g9lwybbndzxrhanblrrk2ndqgdagqqq-glasgow-0-unstable-2025-04-04/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...c70f85c78008b3f1222b000ad65d1d7e468eab03)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-04-20 in filename of file in /nix/store/ry361kmw602whkz03z55yjvscdj51c5p-glasgow-0-unstable-2025-04-20

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/ry361kmw602whkz03z55yjvscdj51c5p-glasgow-0-unstable-2025-04-20 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/a19a706759f213566ce1affbe52052074f16e6cb.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/a19a706759f213566ce1affbe52052074f16e6cb#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/ry361kmw602whkz03z55yjvscdj51c5p-glasgow-0-unstable-2025-04-20
ls -la /nix/store/ry361kmw602whkz03z55yjvscdj51c5p-glasgow-0-unstable-2025-04-20/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for glasgow is: Software for Glasgow, a digital interface multitool

meta.homepage for glasgow is: https://github.com/GlasgowEmbedded/Glasgow




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/GlasgowEmbedded/glasgow/compare/2a67f79d6025a06e98277956cbb036c4237960f1...1b6abc1af7badfa66c19bc356094fd4456d03bdd)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0-unstable-2025-05-01 in filename of file in /nix/store/4bapv1wrfk6g00jghjabn9gz3l817wq9-glasgow-0-unstable-2025-05-01

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

glasgow
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/4bapv1wrfk6g00jghjabn9gz3l817wq9-glasgow-0-unstable-2025-05-01 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A glasgow https://github.com/r-ryantm/nixpkgs/archive/0b1df941527e219d91e13084b79a2be5eff0ad39.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/0b1df941527e219d91e13084b79a2be5eff0ad39#glasgow
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/4bapv1wrfk6g00jghjabn9gz3l817wq9-glasgow-0-unstable-2025-05-01
ls -la /nix/store/4bapv1wrfk6g00jghjabn9gz3l817wq9-glasgow-0-unstable-2025-05-01/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>glasgow</li>
    <li>glasgow.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

flokli commented:
The current version of glasgow in nixpkgs started to fail to build for me:

```
======================================================================
ERROR: test_build (glasgow.applet.video.rgb_input.test.VideoRGBInputAppletTestCase.test_build)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/software/glasgow/applet/video/rgb_input/test.py", line 8, in test_build
    self.assertBuilds(args=["--pins-r", "0:4", "--pins-g", "5:9", "--pins-b", "10:14",
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                            "--pin-dck", "15", "--columns", "160", "--rows", "144",
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                            "--vblank", "960e-6"])
                            ^^^^^^^^^^^^^^^^^^^^^^
  File "/build/source/software/glasgow/applet/__init__.py", line 286, in assertBuilds
    target.build_plan().get_bitstream()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/build/source/software/glasgow/target/hardware.py", line 241, in get_bitstream
    bitstream_data, stdout_data = self.execute(debug=debug)
                                  ~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/build/source/software/glasgow/target/hardware.py", line 193, in execute
    raise GatewareBuildError(
        f"gateware build failed with exit code {proc.returncode}; "
        f"re-run `glasgow` without `-q` to view build log")
glasgow.gateware.GatewareBuildError: gateware build failed with exit code 1; re-run `glasgow` without `-q` to view build log

----------------------------------------------------------------------
Ran 263 tests in 207.986s

FAILED (errors=1, skipped=6)
```

@thoughtpolice, considering https://github.com/GlasgowEmbedded/glasgow/commits/main/'s activity, is there a reasonable commit to pick for a refreshed bump PR?

flokli committed:
glasgow: add myself to maintainers

whitequark commented:
> The software/glasgow/applet/video/rgb_input/test.py test was disabled,
> as it broke with #397611.

You must not do this. Any version of Yosys that crashes on this test is unsuitable for building Amaranth code, and shipping it will result in a broken Glasgow installation. Either roll back Yosys to 0.51, or upgrade Yosys to current HEAD.

flokli commented:
Thanks! I wasn't aware that that specific test is load-bearing, considering a few hundred others succeed :-)

Re-testing this with https://github.com/NixOS/nixpkgs/pull/428579 applied, and the test not disabled.

whitequark commented:
By the way, the Nix derivation should be using the vendored `glasgow.ihex` instead of rebuilding it. We have a [fully reproducible](https://glasgow-embedded.org/latest/develop/firmware.html) firmware build and deploy process which is more rigorous than what the derivation does. I have no intention of supporting arbitrary sdcc versions for building the firmware (we have enough variables as it is) and I would like to know exactly which binary code the device is running at any time.

If you want to rebuild the blob to verify its provenance you could reproduce the contents of the `deploy-firmware.sh` script to build it and then bytewise compare it with the checked-in version, which should pass.

flokli commented:
> By the way, the Nix derivation should be using the vendored `glasgow.ihex` instead of rebuilding it. We have a [fully reproducible](https://glasgow-embedded.org/latest/develop/firmware.html#deploying-firmware) firmware build and deploy process which is more rigorous than what the derivation does. I have no intention of supporting arbitrary sdcc versions for building the firmware (we have enough variables as it is) and I would like to know exactly which binary code the device is running at any time.
> 
> If you want to rebuild the blob to verify its provenance you could reproduce the contents of the `deploy-firmware.sh` script to build it and then bytewise compare it with the checked-in version, which should be identical to the freshly built artifact.


Normally we try to use source builds instead of blobs wherever possible, so ideally we can build the firmware with Nix, and ensure it matches the one checked in the repository as part of the build. I think this can be left for a followup PR though, we aren't using the bundled ihex before this PR either.

whitequark commented:
> Normally we try to use source builds instead of blobs wherever possible, so ideally we can build the firmware with Nix, and ensure it matches the one checked in the repository as part of the build.

Yes, that's completely fine. The reason I flagged a problem here is that people who use Glasgow via Nix would potentially get a blob that's different from what everyone else uses, and I would not know this because I would reasonably assume that the source tree is used as-is.

> I think this can be left for a followup PR though, we aren't using the bundled ihex before this PR either.

Yes, I wanted to inform you of the issue since you're a new maintainer. I'm not suggesting it to be a blocker for this PR.

flokli committed:
glasgow: 0-unstable-2025-01-26 -> 0-unstable-2025-07-28

In addition to bringing all the new features since then, upstream now
uses the same sdcc version, and with an additional normalization phase,
we come up with exactly the same firmware as them (and validate this).

We previously had both the committed firmware blob from upstream, as
well as our built firmware in our output, and only used the upstream one
due to a packaging bug (changed path).

Dependencies were slightly reordered, instead of propagatedBuildInputs
we now use dependencies, and build-system is used instead of
nativeBuildInputs to more align with python package style. I also
documented why pytestCheckHook cannot be used.

flokli committed:
glasgow: remove passthru.updateScript

This cannot be updated automatically, as we need to manually update the
commit ID of the latest commit in the `firmware/` subdirectoy.

flokli commented:
Thanks to lots of debugging with @whitequark, this now produces the same firmware binaries!

I'm not sure we should however merge this as-is, until we managed to iron out the libusb timeouts (TODO: open issue)

flokli committed:
glasgow: simplify PDM_BUILD_SCM_VERSION

Remove one manual step when updating, we don't need to maintain the
commit counter, as discussed in the glasgow IRC channel.
